File: BmiCalculator/app/src/main/java/com/example/bmicalculator/CalculatorActivity.java
package com.example.bmicalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class CalculatorActivity extends AppCompatActivity
{
    private EditText height;
    private EditText weight;
    private Float score;
    private Button calculateButton;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calculator);

        height = (EditText) findViewById(R.id.enterHeightNumber);
        weight = (EditText) findViewById(R.id.enterWeightNumber);

        calculateButton = (Button) findViewById(R.id.CalculateButton);
        calculateButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                scoreActivity();
            }
        });
    }

    public void scoreActivity()
    {
        String heightStr = height.getText().toString();
        String weightStr = weight.getText().toString();

        if (heightStr != null && heightStr != "" && weightStr != null && weightStr != "") {
            float heightValue = Float.parseFloat(heightStr) / 100;
            float weightValue = Float.parseFloat(weightStr);

            score = weightValue / (heightValue * heightValue);
        }

        Intent intent = new Intent(this, ScoreActivity.class);
        intent.putExtra("score", score);
        startActivity(intent);
    }
}

File: BmiCalculator/app/src/main/java/com/example/bmicalculator/CalculatorActivity.kt
package com.example.bmicalculator

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.RadioButton
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity

class CalculatorActivity : AppCompatActivity() {
    private var height: EditText? = null
    private var weight: EditText? = null
    private var age: EditText? = null
    private var radioGroup: RadioGroup? = null
    private var calculateButton: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calculator)
        height = findViewById<View>(R.id.enterHeightNumber) as EditText
        weight = findViewById<View>(R.id.enterWeightNumber) as EditText
        age = findViewById<View>(R.id.enterAgeNumber) as EditText
        radioGroup = findViewById<View>(R.id.radioGroup) as RadioGroup
        calculateButton = findViewById<View>(R.id.CalculateButton) as Button
        calculateButton!!.setOnClickListener { scoreActivity() }
    }

    fun scoreActivity() {
        if (height!!.text.toString() === "" || weight!!.text.toString() === "" || age!!.text.toString() === "") return
        val intent = Intent(this, ScoreActivity::class.java)
        intent.putExtra("height", height!!.text.toString())
        intent.putExtra("weight", weight!!.text.toString())
        intent.putExtra("age", age!!.text.toString())
        val selectedRadioId = radioGroup!!.checkedRadioButtonId
        val selectedRadioButton = findViewById<View>(selectedRadioId) as RadioButton
        intent.putExtra("sex", selectedRadioButton.text)
        startActivity(intent)
    }
}